fix(tui): show in-flight coalesced tool calls in transcript overlay

Chriss4123 opened the pull request:
Fixes #7998.

The TUI coalesces read/search/list exec calls into an in-flight "Exploring/Explored" group (ChatWidget.active_cell). The main viewport renders that active cell immediately, but the Ctrl+T transcript overlay was built only from committed transcript cells, so while the agent is still exploring the overlay can look like the tool calls are missing until a later flush boundary.

This PR makes the transcript overlay include a render-only "live tail" for the current active history cell. It doesn’t change when exec groups flush or how coalescing works; it just makes the overlay reflect what the UI already knows.

To keep overlay interactions cheap, the tail is cached and only recomputed when the active cell changes or the terminal width changes.

Testing:
- cargo test -p codex-tui
- cargo test -p codex-tui2

Chriss4123 committed:
fix: show in-flight coalesced exec calls in transcript overlay

When the TUI coalesces read/search/list exec calls into the in-flight “Exploring/Explored” group, the main viewport renders that active history cell immediately, but the Ctrl+T transcript overlay only shows committed transcript cells. Mid-turn, that can make the transcript look like those tool calls haven’t happened yet until a later flush boundary.

Render the current active history cell as a read-only “live tail” at the bottom of the transcript overlay, without changing any coalescing/flush behavior. Cache the tail and only recompute when the active cell changes or the terminal width changes so scrolling/highlighting in the overlay stays cheap.

Also request a redraw when an exec call completes but the coalesced exec cell remains active, to avoid the transcript “catching up” on an unrelated later event.

Includes tui2 parity.

joshka-oai commented on codex-rs/tui/src/chatwidget.rs:
```diff
@@ -370,6 +371,13 @@ pub(crate) struct ChatWidget {
+    pub(crate) revision: u64,
+    pub(crate) is_stream_continuation: bool,
+    pub(crate) animation_tick: Option<u64>,
+}
```
Add doc comments to help answer 'why does this struct exist' (context / link it back to how it solves this problem).
At the field level - tell me what these track, how they help the solution etc. Why is a revision necessary?

Mindset should be as a developer reading this code for the first time, what do I need to know to orient myself easily against the rest of the code base.

(codex is pretty good at adding this info but generally has to be asked to)

joshka-oai commented on codex-rs/tui/src/pager_overlay.rs:
```diff
@@ -459,21 +470,101 @@ impl TranscriptOverlay {
+        if follow_bottom {
+            self.view.scroll_offset = usize::MAX;
+        }
+    }
```
More docs on how this works needed here - let codex summarize this a bunch

joshka-oai reviewed (changes requested):
Thanks - the idea looks good and it definitely solves the problem.

My comments generally are around adding extra context around the changes to the code. This helps people (and agents) unfamiliar with the specific code paths be able to validate the code actually aligns with what it's intended to do.

I'd merge this with that extra info. Happy to update this for you, or let you go at it if you'd prefer. Mostly codex does a really good job here at adding this info with extra prompts to add the right level of docs (whereeve context / rationale / edge cases / complexity / large abstraction level gaps is needed). Mostly doc comments, some inline.

joshka-oai commented:
@codex review this to ensure that the tui/tui2 behavior and code is the same (modulo any changes that tui2 would prevent this).

Chriss4123 commented on codex-rs/tui/src/chatwidget.rs:
```diff
@@ -2215,6 +2231,10 @@ impl ChatWidget {
     }
 
+    fn bump_active_cell_revision(&mut self) {
+        self.active_cell_revision = self.active_cell_revision.wrapping_add(1);
```
Haha, I think it would take a few hundred billion years under normal circumstances for this to wraparound.

Chriss4123 commented on codex-rs/tui/src/history_cell.rs:
```diff
@@ -100,6 +100,12 @@ pub(crate) trait HistoryCell: std::fmt::Debug + Send + Sync + Any {
+    /// If set, the transcript output is time-dependent and should be re-rendered when the tick
+    /// changes (e.g. for animated spinners/shimmers).
+    fn transcript_animation_tick(&self) -> Option<u64> {
+        None
```
Fixed. `UnifiedExecWaitCell` now emits transcript animation ticks so the shimmer stays live in the overlay.

Chriss4123 commented:
The failing test looks like a flaky timeout. Other platforms passed and this PR only touches `tui`/`tui2`.

Can someone rerun the Windows ARM job?

joshka-oai committed:
docs(tui): clarify transcript overlay live-tail invariants

Document how Ctrl+T transcript overlay stays in sync with the
in-flight active cell via a cached live tail keyed by width,
revision, stream continuation, and animation tick.

Move subsystem rationale into Rustdoc and align module, type, and
function docs between tui and tui2 without changing behavior.
